scrapy_playwright/handler.py

```diff
@@ -39,15 +40,48 @@
     _maybe_await,
 )
 
+# Supporting for Windows
+if sys.platform == "win32" and sys.version_info >= (3, 8):
+    import threading
+
+    class Var:
+        windows_loop = None
+        windows_thread = None
+
+    def windows_get_asyncio_event_loop():
+        if Var.windows_thread is None:
+            if Var.windows_loop is None:
+                Var.windows_loop = asyncio.WindowsProactorEventLoopPolicy().new_event_loop()
+                asyncio.set_event_loop(Var.windows_loop)
+            if not Var.windows_loop.is_running():
+                Var.windows_thread = threading.Thread(
+                    target=Var.windows_loop.run_forever, daemon=True
+                )
+                Var.windows_thread.start()
+        return Var.windows_loop
```

Comment:
According to https://docs.scrapy.org/en/latest/topics/asyncio.html#windows-specific-notes Twisted does not support the Proactor event loop. Is that no longer the case with some new version of Twisted?

Reply:
According to [the Twisted 23.8.0 docs](https://docs.twisted.org/en/twisted-23.8.0/api/twisted.internet.asyncioreactor.AsyncioSelectorReactor.html) this is still the case:
> ...on Python 3.8 and newer the default event loop is asyncio.WindowsProactorEventLoop which is incompatible with [AsyncioSelectorReactor](https://docs.twisted.org/en/twisted-23.8.0/api/twisted.internet.asyncioreactor.AsyncioSelectorReactor.html).

This seems to be running in a different thread, maybe even bypassing the Twisted reactor. I wonder what are the full implications of this, and how desirable it is. The first question that comes to my mind is: do we still need to set the `TWISTED_REACTOR` Scrapy setting or is that irrelevant with this patch?